internal/controllers/configuration/zz_generated_controllers.go

```diff
@@ -999,6 +1018,40 @@ func (r *KongV1KongPluginReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
+	if r.DataplaneClient.AreKubernetesObjectReportsEnabled() {
+		obj.Status.Conditions = nil
+		configuredStatus := r.DataplaneClient.KubernetesObjectIsConfigured(obj)
+		if configuredStatus == k8sobj.ConfiguredStatusSucceeded {
+			obj.Status.Conditions = append(obj.Status.Conditions, metav1.Condition{
+				Type:               kongv1.ProgrammedConditionType,
+				Status:             metav1.ConditionTrue,
+				ObservedGeneration: obj.Generation,
+				Reason:             "ConfiguredInGateway",
+				LastTransitionTime: metav1.Now(),
+			})
+		} else if configuredStatus == k8sobj.ConfiguredStatusFailed {
+			obj.Status.Conditions = append(obj.Status.Conditions, metav1.Condition{
+				Type:               kongv1.ProgrammedConditionType,
+				Status:             metav1.ConditionFalse,
+				ObservedGeneration: obj.Generation,
+				Reason:             "ConfigurationFailed",
+				LastTransitionTime: metav1.Now(),
+			})
+		} else {
+			obj.Status.Conditions = append(obj.Status.Conditions, metav1.Condition{
+				Type:               kongv1.ProgrammedConditionType,
+				Status:             metav1.ConditionUnknown,
+				ObservedGeneration: obj.Generation,
+				Reason:             "Unknown",
+				LastTransitionTime: metav1.Now(),
+			})
+		}
+
+		if err := r.Status().Update(ctx, obj); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
```

Comment:
Reading `configuredStatus := r.DataplaneClient.KubernetesObjectIsConfigured(obj)` we can tell if the object:
1) was successfully configured in Gateway 
2) failed to be configured in Gateway
3) its state is unknown, most probably during the first time controller sees the object

Reply:
Please note that no specific message is set for `ConfigurationFailed` reason. It could be added by passing it via `KongClient.kubernetesObjectReportsFilter`. 

The question is whether we want to present the same messages both in Events and Conditions. Options are:
1) Duplicate the detailed information regarding specific parsing errors in `Event.Message` and `Condition.Message` 
2) Fill in just `Event.Message` and leave the `Condition` with a generic `ConfigurationFailed` reason. The events are going to be bound to the objects so it's not gonna be complex to get this information from the user's perspective anyway.